app/api/users.py

```diff
@@ -216,11 +216,13 @@ def before_update_object(self, user, data, view_kwargs):
         if has_access('is_super_admin') and data.get('is_admin') and data.get('is_admin') != user.is_admin:
             user.is_admin = not user.is_admin
 
-        if has_access('is_admin') and data.get('is_sales_admin') != user.is_sales_admin and ('is_sales_admin' in data):
-            user.is_sales_admin = not user.is_sales_admin
+        if 'is_sales_admin' in data:
+            if has_access('is_admin') and data.get('is_sales_admin') != user.is_sales_admin:
```

Comment:
No need to use nested if, just use and operator

Reply:
Like in current commit?